contentctl/objects/savedsearches_conf.py

```diff
@@ -0,0 +1,187 @@
+
+    def model_post_init(self, __context: Any) -> None:
+        super().model_post_init(__context)
+        self._parse_detection_stanzas()
+
+    def is_section_header(self, line: str) -> bool:
+        """
+        Given a line, determine if the line is a section header, indicating the start of a new
+        section
+        :param line: a line from the conf file
+        :returns: a bool indicating whether the current line is a section header or not
+        """
+        # Compile the pattern based on the app name
+        pattern = re.compile(r"\[" + self.app_label + r" - .+ - Rule\]")
+        if pattern.match(line):
+            return True
+        return False
+
+    def section_start(self, line: str) -> None:
+        """
+        Given a line, adjust the state to track a new section
+        :param line: a line from the conf file
+        """
+        # Determine the new section name:
+        new_section_name = line.strip().strip("[").strip("]")
+
+        # Raise if we are in a section already according to the state (we cannot statr a new section
+        # before ending the previous section)
+        if self._in_section:
+            raise Exception(
+                "Attempting to start a new section w/o ending the current one; check for "
+                f"parsing/serialization errors: (current section: '{self._current_section_name}', "
+                f"new section: '{new_section_name}') [see line {self._current_line_no} in "
+                f"{self.path}]"
+            )
+
+        # Capture the name of this section, reset the lines, and indicate that we are now in a
+        # section
+        self._current_section_name = new_section_name
+        self._current_section_lines = [line]
+        self._in_section = True
+
+    def section_end(self) -> None:
+        """
+        Adjust the state end the section we were enumerating; parse the lines as a DetectionStanza
+        """
+        # Name should have been set during section start
+        if self._current_section_name is None:
+            raise Exception(
+                "Name for the current section was never set; check for parsing/serialization "
+                f"errors [see line {self._current_line_no} in {self.path}]."
+            )
+        elif self._current_section_name in self.detection_stanzas:
+            # Each stanza should be unique, so the name should not already be in the dict
+            raise Exception(
+                f"Name '{self._current_section_name}' already in set of stanzas [see line "
+                f"{self._current_line_no} in {self.path}]."
+            )
+
+        # Build the stanza model from the accumulated lines and adjust the state to end this section
+        self.detection_stanzas[self._current_section_name] = DetectionStanza(
+            name=self._current_section_name,
+            lines=self._current_section_lines
+        )
+        self._in_section = False
+
+    def _parse_detection_stanzas(self) -> None:
```

Comment:
Note that I had to bake my custom parser here, since the existing Python configparser and the open source Python for Splunk conf parser both did not work (both do to an inability to parse the file, and a lack of necessary features)

Reply:
I think this is also just included for information, so I will mark as resolved :)